pkg/cmd/roachtest/test_impl.go

```diff
@@ -195,6 +195,11 @@ func (t *testImpl) StandardCockroach() string {
 }
 
 func (t *testImpl) DeprecatedWorkload() string {
+	// Fail and give a hint that the test should set `RequiresDeprecatedWorkload` so
+	// we can validate the binary exists.
+	if !t.spec.RequiresDeprecatedWorkload {
```

Comment:
Can we just fail here if `deprecatedWorkload` is "" instead of introducing "RequiresDeprecatedWorkload"?

Reply:
My comment is badly worded, I'll update it. I'm not guarding against the binary not existing, I'm guarding against people using the deprecated workload when not needed.

In a previous PR, I refactored almost all the `DeprecatedWorkload()` calls to just use `cockroach workload` instead. I want future tests to stop using `DeprecatedWorkload` if possible, so if a test does need it I want the test to explicitly set `RequiresDeprecatedWorkload`. 

If we just check that a `deprecatedWorkload` exists, it would be possible for a test writer to not set this spec and continue using the `deprecatedWorkload` without even knowing it's discouraged.

Let me know if that makes sense.